opendcs-rest-api/src/main/java/org/opendcs/odcsapi/sec/basicauth/BasicAuthResource.java

```diff
@@ -277,11 +275,16 @@ private Set<OpenDcsApiRoles> getUserRoles(String username)
 
 	private ApiAuthorizationDAI getAuthDao()
 	{
-		TimeSeriesDb timeSeriesDb = getLegacyTimeseriesDB();
+		DataSource dataSource = getDataSource();
+		String databaseType = getDatabaseType(dataSource);
 		// Username+Password login only supported by OpenTSDB
-		if(timeSeriesDb.isOpenTSDB())
+		if("opentsdb".equalsIgnoreCase(databaseType))
+		{
+			return new OpenTsdbAuthorizationDAO(dataSource);
+		}
+		else if("cwms".equalsIgnoreCase(databaseType))
 		{
-			return new OpenTsdbAuthorizationDAO(timeSeriesDb);
+			return new CwmsAuthorizationDAO(dataSource);
```

Comment:
Note that once https://github.com/opendcs/opendcs/pull/1525 is merged in we should move to those systems.

(just a warning, this seems a reasonable workaround for the moment.)